bladex/propeller.py

```diff
@@ -10,62 +10,83 @@
+        blade_solid = blade.generate_blade_solid(max_deg=2,
                                                  display=False,
                                                  errors=None)
         blades = []
         blades.append(blade_solid)
-        for i in range(self.number_of_blades-1):
-            rot_dir = gp_Dir(1.0,0.0,0.0)
-            origin = gp_Pnt(0.0,0.0,0.0)        
+        for i in range(n_blades-1):
+            rot_dir = gp_Dir(1.0, 0.0, 0.0)
+            origin = gp_Pnt(0.0, 0.0, 0.0)        
             rot_axis = gp_Ax1(origin, rot_dir)
             transf = gp_Trsf()
-            transf.SetRotation(rot_axis, float(i+1)*2.0*np.pi/float(self.number_of_blades));
+            transf.SetRotation(rot_axis, float(i+1)*2.0*np.pi/float(n_blades));
             transformer = BRepBuilderAPI_Transform(blade_solid, transf);
             blades.append(transformer.Shape())
         blades_combined = blades[0]
         for i in range(len(blades)-1):
-            boolean_union = BRepAlgoAPI_Fuse(blades_combined,blades[i+1])
+            boolean_union = BRepAlgoAPI_Fuse(blades_combined, blades[i+1])
             boolean_union.Build()
             if not boolean_union.IsDone():
                 raise RuntimeError('Unsuccessful assembling of blade')
             blades_combined = boolean_union.Shape()
-        boolean_union = BRepAlgoAPI_Fuse(shaft_solid, blades_combined)
+        boolean_union = BRepAlgoAPI_Fuse(self.shaft_solid, blades_combined)
         boolean_union.Build()
-        result_solid = boolean_union.Shape()
+        result_compound = boolean_union.Shape()
         section_edges = boolean_union.SectionEdges()
         sewer = BRepBuilderAPI_Sewing(1e-2)
-        sewer.Add(result_solid)
+        sewer.Add(result_compound)
         sewer.Perform()
-        sewed_full_body = sewer.SewedShape()
+        self.sewed_full_body = sewer.SewedShape()
+
+    def generate_propeller_iges(self, propeller_and_shaft, display=False):
+        """
+        Export and plot the .iges CAD for the propeller with shaft.
+        This method requires PythonOCC to be installed.
+
+        :raises RuntimeError: if the solid assembling of blades is not 
+            completed successfully
+        :param string propeller_and_shaft: path where to store the .iges CAD 
+            for the propeller and shaft
+        :param bool display: if True, then display the propeller with shaft. 
+            Default value is False
+        """
         iges_writer = IGESControl_Writer()
-        iges_writer.AddShape(sewed_full_body)
+        iges_writer.AddShape(self.sewed_full_body)
         iges_writer.Write(propeller_and_shaft + '.iges')
-        write_stl_file(sewed_full_body, propeller_and_shaft + '.stl')
\ No newline at end of file
+        if display:
+            display, start_display, add_menu, add_function_to_menu = init_display()
+            display.DisplayShape(self.sewed_full_body, update=True)
+            start_display()
```

Comment:
Why display when exporting? It looks much better to me have a separate method to display the propeller like>
```
my_custom_propeller = Propeller(...)
my_custom_propeller.display()
```

Reply:
I can change accordingly, thanks.